lte/protos/subscriberdb.proto

```diff
@@ -213,14 +218,50 @@ message SubscribersDigest {
   string md5_base64_digest = 1;
 }
 
+message QueryFlatDigestRequest {
+  // digest is the deterministic digest of the full set of subscriber objects
+  // stored on the client side.
+  SubscribersDigest digest = 1;
+}
+
+message QueryFlatDigestResponse {
+  // no_updates is true if client's existing subscribers match those on the server.
+  bool no_updates = 1;
+  // digest is the deterministic flat digest of the full set of subscriber objects
+  // defined by the server side.
+  SubscribersDigest digest = 2;
+}
+
+message QueryPerSubscriberDigestRequest {
+  // per_sub_digest is a set of per-subscriber digests, indexed by the
+  // subscriber IDs (IMSI).
+  map<string, SubscribersDigest> per_sub_digest = 1;
+}
+
+message SubscriberDataWithDigest {
+  // digest is the deterministic digest of a single subscriber.
+  SubscribersDigest digest = 1;
+  // data is the original subscriber data.
+  SubscriberData data = 2;
+}
+
+message QueryPerSubscriberDigestResponse {
```

Comment:
This isn't how I'd structure this message

- added/removed can just be a list of IMSIs
- modified can be a map of IMSI->SubscriberData
- the full/updated map of IMSI->SubscriberDigest can be its own field (normalized)
- will also need to include the new flat digest

Reply:
Wouldn't `added` also be a map of IMSI -> `SubscriberData` since the client doesn't have that data yet? In which case would this structure make sense:
```
bool resync = 1;
SubscribersDigest flat_digest = 2;
map<string, SubscribersDigest> per_sub_digests = 3;
map<string, SubscribersDataWithDigest> added_modified = 4;
repeated string removed = 5;
```
@hcgatewood